### PR TITLE
Further changes to ParameterHandler::parse_input_from_json

### PR DESCRIPTION
**Pull request by peterrum**
This PR builds up on #9430. Only the last commit is relevant for this PR. I will rebase this branch once the other PR is merged.

This PR enables to parse the following .json-file:
```json
{
    "int1": 1,
    "int2": 2
}
```
with the following parameter declaration:
```
  prm.add_parameter("int1", int1);
```
Please not that I do not require to write `"int1": {"value" : 1}` and I was able to skip undefined parameter entries.

These changes are useful for two reasons. Reason 1) by not having to write `:{"value" : X}` the .json-file becomes significantly shorter and more user friendly (which is why I made these changes a long time ago on my own fork); 2) by skipping undefined sections/entries you can have multiple `ParameterHandler`s operating on the same file (maybe at different stages of the program).

ping @luca-heltai 

FYI @lpsaavedra

### CODE REVIEW

**Comment by peterrum** [comment]
BTW: I like to use .json file because it is well supported by Python!

File: doc/news/changes/minor/20200127PeterMunch-1 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+New: ParameterHandler::parse_input_from_json and  ParameterHandler::parse_input_from_xml can skip
```
Comment (kronbichler) [inline doc/news/changes/minor/20200127PeterMunch-1]:
```suggestion
New: ParameterHandler::parse_input_from_json() and  ParameterHandler::parse_input_from_xml() can skip
```
We try to consistently use the braces in function names (even though they are not necessary for doxygen to resolve them).

File: doc/news/changes/minor/20200127PeterMunch-2 [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Improved: ParameterHandler::parse_input_from_json can handle values of type `"a" : 1', in addition to
```
Comment (kronbichler) [inline doc/news/changes/minor/20200127PeterMunch-2]:
```suggestion
Improved: ParameterHandler::parse_input_from_json() can handle values of type `"a" : 1', in addition to
```

**Comment by peterrum** [comment]
@kronbichler I have added the braces!

**Comment by peterrum** [comment]
I have rebased with master and removed the merge conflicts! Ready to merge!

**Pull request merged**